### PR TITLE
testcase/le_tc: Add libc testcase

### PR DESCRIPTION
**Pull request by jeongchanKim**
strtof and strtold convert a string to float/long double value.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [05c3e7122dfdc1de1aafb3c58c4dcea3ee5104da] -

**Comment by seinfra** [comment]
Target : [2ab6104b0651ad2111caf848661092a926ec8afe] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2ab6104b0651ad2111caf848661092a926ec8afe] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_string.c [hunk 1]
```diff
@@ -850,6 +850,56 @@ static void tc_libc_string_strlcpy(void)
 
 	TC_SUCCESS_RESULT();
 }
+/**
+* @fn                   :tc_libc_string_strtof
+* @brief                :conver the string to float value
```
Comment (manoharahk) [inline apps/examples/testcase/le_tc/kernel/tc_libc_string.c]: conver -> convert ?

**Comment by seinfra** [comment]
Target : [296841c92aef4ce818ec0ac92c8c2f1386fb96f3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [296841c92aef4ce818ec0ac92c8c2f1386fb96f3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2c38f3599d1688c9818119c9226be48ef03286f1] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_math.c [hunk 2]
```diff
@@ -3529,6 +3582,8 @@ int libc_math_main(void)
 	tc_libc_math_y1f();
 	tc_libc_math_yn();
 	tc_libc_math_ynf();
+	tc_libc_math_gamma();
```
Comment (manoharahk) [inline apps/examples/testcase/le_tc/kernel/tc_libc_math.c]: Should it be added in alphabetical order?

File: apps/examples/testcase/le_tc/kernel/tc_libc_math.c [hunk 3]
```diff
@@ -3395,6 +3395,59 @@ static void tc_libc_math_ynf(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+ * @fn                   :tc_libc_math_gammna
```
Comment (manoharahk) [inline apps/examples/testcase/le_tc/kernel/tc_libc_math.c]: Should it be gammna  -> gamma?

File: apps/examples/testcase/le_tc/kernel/tc_libc_math.c [hunk 4]
```diff
@@ -3395,6 +3395,59 @@ static void tc_libc_math_ynf(void)
...
+ * @return               :void
+ */
+static void tc_libc_math_lgamma(void)
+{
+	double value;
+	double result;
+
+	value = 2.0;
+	result = lgamma(value);
+	TC_ASSERT_EQ("gamma", result, 0);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_math.c]: lgamma

File: apps/examples/testcase/le_tc/kernel/tc_libc_math.c [hunk 5]
```diff
@@ -3395,6 +3395,59 @@ static void tc_libc_math_ynf(void)
...
+
+	value = INFINITY;
+	result = lgamma(value);
+	TC_ASSERT_EQ("gamma", result, (double)INFINITY);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_math.c]: lgamma

**Comment by seinfra** [comment]
Target : [d7e126c557c9a71426e544624a52e36d7baa750e] -

**Comment by seinfra** [comment]
Target : [d7e126c557c9a71426e544624a52e36d7baa750e] -

File: apps/examples/testcase/le_tc/kernel/tc_libc_string.c [hunk 6]
```diff
@@ -850,6 +850,56 @@ static void tc_libc_string_strlcpy(void)
...
+* @return               :void
+*/
+static void tc_libc_string_strtold(void)
+{
+	char *str;
+	char *ptr;
+	long double value;
+
+	str = "123.456TizenRT";
+	value = strtold(str, &ptr);
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/kernel/tc_libc_string.c]: what about returned 'value' check?

File: apps/examples/testcase/le_tc/kernel/tc_libc_string.c [hunk 7]
```diff
@@ -850,6 +850,56 @@ static void tc_libc_string_strlcpy(void)
...
+	TC_ASSERT_EQ("strtold", strncmp(ptr, "TizenRT", strlen("TizenRT")), 0);
+
+	str = "-78.9123TinyAra";
+	value = strtold(str, &ptr);
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/kernel/tc_libc_string.c]: same as above

**Comment by seinfra** [comment]
Target : [9920d865ee9e80932471b8b2c28f62939a2d0f03] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9920d865ee9e80932471b8b2c28f62939a2d0f03] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_string.c (hunk 6 above)
Comment (jeongchanKim) [reply apps/examples/testcase/le_tc/kernel/tc_libc_string.c]: modified

**Comment by seinfra** [comment]
Target : [a7d9dfbcf73485cfa3d0c1b98008c53d42778c76] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a7d9dfbcf73485cfa3d0c1b98008c53d42778c76] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7b99302df7af8d06c8f3b454a4a90248987c954b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7b99302df7af8d06c8f3b454a4a90248987c954b] - Code Rule Check OK.

**Pull request merged**